src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts

```diff
@@ -114,6 +115,17 @@ export class CodeEditorTutorAssessmentContainerComponent implements OnInit, OnDe
                     this.automaticResult = this.getLatestAutomaticResult(this.participation.results);
                     this.manualResult = this.getLatestManualResult(this.participation.results);
 
+                    // Setup automatic feedback, set or round credits and set detailText if not already set
+                    this.automaticFeedback = this.automaticResult?.feedbacks!;
+                    this.automaticFeedback.forEach((feedback) => {
+                        feedback.id = undefined;
+                        if (feedback.credits) {
+                            feedback.credits = Math.round(feedback.credits);
+                        } else {
+                            feedback.credits = 0;
```

Comment:
This is the score/credits for failed test cases. I added this here as I was not sure if it is better to add it somewhere at the `updateResult` method of `ProgrammingExerciseGradingService.java`

Reply:
We could set the credits by default to 0 when the feedback is created, however the client check would still be needed for existing feedback / existing exercises so i think this is fine.

Reply:
But why are you rounding the credits??

Reply:
@schultek I rounded it because I got e.g. 0,97884545  for a positive automatic feedback

Reply:
But if we round the credits per automatic feedback, it no longer obeys the grading scheme defined in the test case and SCA section. This would make `0.51...` into `1.0` and `0.49...` into `0.0`, right? I think this does not reflect fair grading and rounding already caused us enough trouble during the PGdP and EidI exams. My recommendation would be to just leave it as precise as possible, and maybe only display it better (or cut off / round up at some place far behind the dot).
I don't think this will be much of a problem because very often the programming exercises never had "nice" credit number.

Reply:
Yes rounding would lead to a wrong result, but as @MaisiKoleni suggested you could just display a rounded number.

Reply:
Done